src/PubSubClient.cpp

```diff
@@ -236,21 +236,21 @@ bool PubSubClient::connected() {
         DEBUG_PSC_PRINTF("lost connection (client may have more details)\n");
         _state = MQTT_CONNECTION_LOST;
         _client->stop();
-        pingOutstanding = false;
+        _pingOutstanding = false;
     }
     return false;
 }
 
 void PubSubClient::disconnect() {
     DEBUG_PSC_PRINTF("disconnect called\n");
-    this->buffer[0] = MQTTDISCONNECT;
-    this->buffer[1] = 0;
-    _client->write(this->buffer, 2);
+    _buffer[0] = MQTTDISCONNECT;
+    _buffer[1] = 0;
+    _client->write(_buffer, 2);
     _state = MQTT_DISCONNECTED;
     _client->flush();
     _client->stop();
```

Comment:
Not sure if you can call this with no `_client` set, but it feels a bit better if there's a check to see if it is a valid pointer before dereferencing it... ;)